Classes/Event/EventNormalizer.php

```diff
@@ -0,0 +1,57 @@
+<?php
+namespace Neos\EventSourcing\Event;
+
+/*
+ * This file is part of the Neos.EventSourcing package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+use Neos\Flow\Annotations as Flow;
+use Symfony\Component\Serializer\Normalizer\DateTimeNormalizer;
+use Symfony\Component\Serializer\Normalizer\JsonSerializableNormalizer;
+use Symfony\Component\Serializer\Normalizer\ObjectNormalizer;
+use Symfony\Component\Serializer\Serializer;
+
+/**
+ * @Flow\Scope("singleton")
+ *
+ * Converts instances of EventInterface to serializable arrays vice versa
+ */
+final class EventNormalizer
+{
+
+    /**
+     * @Flow\Inject
+     * @var EventTypeResolver
+     */
+    protected $eventTypeResolver;
+
+    /**
+     * @var Serializer
+     */
+    private $serializer;
+
+    protected function initializeObject(): void
+    {
+        // TODO: make normalizers configurable
+        $normalizers = [new DateTimeNormalizer(\DateTime::ISO8601), new JsonSerializableNormalizer(), new ObjectNormalizer()];
```

Comment:
What about `DateIntervalNormalizer`? We do support DateInterval in our current EventToArray TypeConverter. Also `ArrayDenormalizer` would add support for `SomeClass[]` properties, e.g. when an event contains a list of some VOs.

Also, `GetSetMethodNormalizer` is supposedly faster than `ObjectNormalizer` and should be used if the objects serialized contain getters (and possibly setters).

See https://symfony.com/doc/current/serializer.html#adding-normalizers-and-encoders

Reply:
Yes, absolutely! This is merely a first WIP with some hard-coded normalizers to test the basics. This got to be configurable anyways (see TODO comment)